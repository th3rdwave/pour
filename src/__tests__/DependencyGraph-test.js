// @flow

const DependencyGraph = require('../DependencyGraph');

describe('DependencyGraph', () => {
  it('creates a simple graph', () => {
    const config = {
      projects: [
        {
          name: 'app',
          dependencies: ['shared'],
        },
        {
          name: 'api',
          dependencies: ['shared'],
        },
        {
          name: 'shared',
        },
      ],
    };

    const graph = DependencyGraph.create(config);
    expect(graph).toMatchSnapshot();
  });

  it('traverses a graph', () => {
    const config = {
      projects: [
        {
          name: 'app',
          dependencies: ['shared'],
        },
        {
          name: 'api',
          dependencies: ['shared'],
        },
        {
          name: 'shared',
        },
      ],
    };

    const graph = DependencyGraph.create(config);

    const cb = jest.fn().mockReturnValue(true);
    DependencyGraph.traverse(cb, graph);
    expect(cb).toHaveBeenCalledTimes(5);
  });

  it('maps once a graph', () => {
    const config = {
      projects: [
        {
          name: 'app',
          dependencies: ['shared'],
        },
        {
          name: 'api',
          dependencies: ['shared'],
        },
        {
          name: 'shared',
        },
      ],
    };

    const graph = DependencyGraph.create(config);

    const cb = jest.fn(e => e.node.name);
    const res = DependencyGraph.mapOnce(cb, graph);
    expect(cb).toHaveBeenCalledTimes(3);
    expect(res).toMatchSnapshot();
  });

  it('removes all nodes by name', () => {
    const config = {
      projects: [
        {
          name: 'app',
          dependencies: ['shared'],
        },
        {
          name: 'api',
          dependencies: ['shared'],
        },
        {
          name: 'shared',
        },
      ],
    };

    const graph = DependencyGraph.create(config);
    DependencyGraph.remove('shared', graph);
    expect(graph).toMatchSnapshot();
  });
});
