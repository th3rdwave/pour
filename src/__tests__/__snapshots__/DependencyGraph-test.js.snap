// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyGraph creates a simple graph 1`] = `
Object {
  "edges": Array [
    Object {
      "edges": Array [
        Object {
          "edges": Array [],
          "node": Object {
            "name": "shared",
          },
        },
      ],
      "node": Object {
        "dependencies": Array [
          "shared",
        ],
        "name": "app",
      },
    },
    Object {
      "edges": Array [
        Object {
          "edges": Array [],
          "node": Object {
            "name": "shared",
          },
        },
      ],
      "node": Object {
        "dependencies": Array [
          "shared",
        ],
        "name": "api",
      },
    },
    Object {
      "edges": Array [],
      "node": Object {
        "name": "shared",
      },
    },
  ],
  "node": Object {
    "name": "root",
  },
}
`;

exports[`DependencyGraph maps once a graph 1`] = `
Array [
  "shared",
  "api",
  "app",
]
`;

exports[`DependencyGraph removes all nodes by name 1`] = `
Object {
  "edges": Array [
    Object {
      "edges": Array [],
      "node": Object {
        "dependencies": Array [
          "shared",
        ],
        "name": "app",
      },
    },
    Object {
      "edges": Array [],
      "node": Object {
        "dependencies": Array [
          "shared",
        ],
        "name": "api",
      },
    },
  ],
  "node": Object {
    "name": "root",
  },
}
`;
